### PR TITLE
Barchart: Adds the ability to set color of clicked bar 

### PR DESCRIPTION
**Pull request by tumentumurchudur**
This PR allows users of the `BarChart` component to be able to set the color of the clicked bar by passing a prop `colorWhenClicked`.

ezgif com-video-to-gif 1

### CODE REVIEW

File: src/components/BarChart.js [hunk 1]
```diff
@@ -175,12 +207,14 @@ const BarChart = React.createClass({
       return (
         <Rect
           animateOnHover={this.props.animateOnHover}
+          clickedRect={this.state.clickedRect}
           color={this.props.color}
+          colorWhenClicked={this.props.colorWhenClicked}
           height={height}
           hoverColor={this.props.hoverColor}
           key={index}
```
Comment (redperk) [inline src/components/BarChart.js]: Is there something unique we can change this to?

Comment (tumentumurchudur) [inline src/components/BarChart.js]: Yeah, let me address that.

**Comment by tumentumurchudur** [comment]
@redperk Your concern with `index` being used for key is addressed. Ready for another review.

**Comment by redperk** [comment]
:shipit:

**Pull request merged**